bin/repos-lint

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+
+import sys
+import re
+import yaml
+import socket
+from urllib.parse import urlparse
+from urllib.request import urlopen
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("filename")
+args = parser.parse_args()
+
+product_pattern = re.compile('^[A-z_0-9]+$')
+
+def validate_product(pkg):
+    print('%16s: ' % 'product name', end='')
+    try:
+        validate_string(pkg)
+        validate_product_name(pkg)
+    except ValueError as e:
+        print(e)
+        return False
+
+    print('OK')
+    return True
+
+def validate_string(s):
+    if not type(s) is str:
+        raise ValueError('is not a string')
+
+def validate_product_name(pkg):
+    if not product_pattern.match(pkg):
+        raise ValueError('invalud EUPS product name')
+
+def validate_repo(repo):
+    print('%16s: ' % 'repo url', end='')
+
+    try:
+        validate_string(repo)
+        url = urlparse(repo)
+
+        if url.scheme == 'https':
+            urlopen(repo)
+        elif url.scheme == 'git':
+            # don't want to attempt to clone the repo as it might be large, so
+            # we're just checking for a valid hostname.  A better check might
+            # be to attempt a raw tcp connection.
+            socket.gethostbyname(url.netloc)
+        else:
+            raise ValueError('unsuported scheme: %s' % url.scheme)
+    except Exception as e:
+        print(e)
+        return False
+
+    print('OK')
+    return True
+
+repos=0
+problemos=0
+
+print('repos')
+print('===')
+print()
+
+with open(args.filename, 'r') as f:
+    data = yaml.safe_load(f)
+    if not type(data) is dict:
+        print('invalid format: must be a hash')
```

Comment:
Actually, it's YAML speak.  Technically, it's a mapping but the common usage seems to be hash.